web-admin/src/features/projects/status/ProjectResources.svelte

```diff
@@ -2,54 +2,154 @@
     V1ReconcileStatus,
+    type V1Resource,
+    type V1ListResourcesResponse,
   } from "@rilldata/web-common/runtime-client";
   import { runtime } from "@rilldata/web-common/runtime-client/runtime-store";
   import { useQueryClient } from "@tanstack/svelte-query";
   import Button from "web-common/src/components/button/Button.svelte";
   import ProjectResourcesTable from "./ProjectResourcesTable.svelte";
-  import RefreshConfirmDialog from "./RefreshConfirmDialog.svelte";
+  import RefreshAllSourcesAndModelsConfirmDialog from "./RefreshAllSourcesAndModelsConfirmDialog.svelte";
+  import { eventBus } from "@rilldata/web-common/lib/event-bus/event-bus";
+  import { useResources } from "./selectors";
+  import { ResourceKind } from "@rilldata/web-common/features/entity-management/resource-selectors";
+  import { onNavigate } from "$app/navigation";
+  import { onMount } from "svelte";
 
   const queryClient = useQueryClient();
   const createTrigger = createRuntimeServiceCreateTrigger();
 
+  let isConfirmDialogOpen = false;
+  let isPollingEnabled = false;
+  let currentResourceName: string | undefined;
   let isReconciling = false;
-  let isRefreshConfirmDialogOpen = false;
+  let isLoaded = false;
+
+  const INITIAL_POLL_INTERVAL = 1_000;
+  const MAX_POLL_INTERVAL = 10_000;
+  const BACKOFF_THRESHOLD_MS = 30_000;
+  let pollStartTime: number | null = null;
 
   $: ({ instanceId } = $runtime);
 
-  $: resources = createRuntimeServiceListResources(
-    instanceId,
-    // All resource "kinds"
-    undefined,
-    {
-      query: {
-        select: (data) => {
-          // Filter out the "ProjectParser" resource
-          return data.resources.filter(
-            (resource) =>
-              resource.meta.name.kind !== "rill.runtime.v1.ProjectParser",
-          );
-        },
-        refetchOnMount: true,
-        refetchOnWindowFocus: true,
-        refetchInterval: isReconciling ? 500 : false,
-      },
-    },
-  );
+  function isResourceErrored(resource: V1Resource) {
+    return resource.meta.reconcileError;
+  }
 
-  $: isAnySourceOrModelReconciling = $resources?.data?.some(
-    (resource) =>
+  function isResourceReconciling(resource: V1Resource) {
+    return (
       resource.meta.reconcileStatus ===
         V1ReconcileStatus.RECONCILE_STATUS_PENDING ||
       resource.meta.reconcileStatus ===
-        V1ReconcileStatus.RECONCILE_STATUS_RUNNING,
+        V1ReconcileStatus.RECONCILE_STATUS_RUNNING
+    );
+  }
+
+  $: resources = useResources(instanceId, {
+    select: (data: V1ListResourcesResponse) => ({
+      ...data,
+      // Filter out project parser and refresh triggers
+      resources: data?.resources?.filter(
+        (resource: V1Resource) =>
+          resource.meta.name.kind !== ResourceKind.ProjectParser &&
+          resource.meta.name.kind !== ResourceKind.RefreshTrigger,
+      ),
+    }),
+    refetchInterval: (data) => {
+      if (
+        !isPollingEnabled ||
+        $resources?.isError ||
```

Comment:
This looks circular. The `refetchInterval` function should be a pure function that only cares about the `data`.